cmd/earthly/subcmd/common.go

```diff
@@ -22,6 +25,36 @@ func (b *byteSizeValue) Set(s string) error {
 
 func (b *byteSizeValue) String() string { return humanize.Bytes(uint64(*b)) }
 
+// duration implements cli.GenericFlag methods to support time.Duration with days, e.g. 1d
+type duration struct {
```

Comment:
nit: is it possible to just do `type duration time.Duration` instead?

Reply:
also this feels like something that should exist in a seperate package, possibly under `util/timeutil` ?

Reply:
I'm not sure I'll be able to pass the value from `Set(value string)`, or am I missing something?

Reply:
oh, actually looks like I could:
`*d = duration(value)`

Reply:
haha beat me to it, I just put this very simple example together:

```
package main

import (
	"fmt"
	"time"
)

type duration time.Duration

func (d *duration) Set(value string) {
	fmt.Printf("hello %s\n", value)
	*d = duration(time.Second)
}

func main() {
	var d duration
	fmt.Printf("before: %v\n", d)
	d.Set("world")
	fmt.Printf("after: %v\n", d)
}
```